mvcc/key_index.go

```diff
@@ -210,18 +223,21 @@ func (ki *keyIndex) compact(atRev int64, available map[revision]struct{}) {
 
 	if !g.isEmpty() {
 		n := g.walk(f)
-		// remove the previous contents.
-		if n != -1 {
-			g.revs = g.revs[n:]
-		}
 		// remove any tombstone
-		if len(g.revs) == 1 && i != len(ki.generations)-1 {
+		if len(g.revs)-1 == n && i != len(ki.generations)-1 {
```

Comment:
yeah, the code should be `delete(available, g.revs[n])` instead of `delete(available, g.revs[0])`.